pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -735,12 +762,9 @@ func findRegistryWithParsedConfig(config *parsedConfig, ref string) (*Registry,
 	reg := Registry{}
 	prefixLen := 0
 	for _, r := range config.partialV2.Registries {
-		if refMatchesPrefix(ref, r.Prefix) {
-			length := len(r.Prefix)
-			if length > prefixLen {
-				reg = r
-				prefixLen = length
-			}
+		if refMatchesPrefix(ref, r.Prefix) > 0 {
+			reg = r
+			prefixLen = len(r.Prefix)
```

Comment:
This looks incorrect as we're now ignoring the length of the prefix and would always select the "last" matching prefix rather than the longest one.

Consider the following examplery config:
```toml
[[registry]]
location="1.com"
prefix="foo.com/example"
[[registry]]
location="2.com"
prefix="*.foo.com/example"
[[registry]]
location="3.com"
prefix="*.com/example"
```

Now, we do a `podman pull sub.foo.com/example:latest`.  With the current change, we'd pull from "3.com" but we should be pulling from "2.com" since it has the longest matching prefix.

Reply:
ack